benchmarks/definition/regression/core.py

```diff
@@ -0,0 +1,494 @@
+                    0, settings.noise_std, len(target_train)
+                )
+
+            # Evaluate naive models once per (mc_iter, n_train_pts)
+            naive_results = _evaluate_naive_models(
+                target_train,
+                target_test,
+                vanilla_searchspace,
+                tl_searchspace,
+                objective,
+            )
+
+            # Add metadata for naive models (fraction_source = 0 for consistency)
+            for naive_result in naive_results:
+                naive_result.update(
+                    {
+                        "mc_iter": mc_iter,
+                        "n_train_pts": n_train_pts,
+                        "fraction_source": 0.0,  # Naive models don't use source data
+                        "n_source_pts": 0,
+                        "n_test_pts": len(target_test),
+                        "source_data_seed": settings.random_seed + mc_iter,
+                    }
+                )
+                results.append(naive_result)
+
+            # Evaluate transfer learning models for each source fraction
+            for fraction_source in settings.source_fractions:
+                # Update progress bar description
+                pbar.set_description(
+                    f"MC {mc_iter + 1}/{settings.n_mc_iterations} | "
+                    f"Frac {fraction_source:.2f} | "
+                    f"Pts {n_train_pts}/{settings.max_n_train_points}"
+                )
+
+                # Sample source data using configured sampling strategy
+                source_subset = _sample_source_data(
+                    source_data,
+                    source_tasks,
+                    fraction_source,
+                    name_task,
+                    settings.random_seed + mc_iter,
+                    settings.stratified_source_sampling,
+                )
+
+                tl_results = _evaluate_transfer_learning_models(
+                    source_subset,
+                    target_train,
+                    target_test,
+                    tl_searchspace,
+                    objective,
+                    fraction_source,
+                )
+
+                # Add metadata for transfer learning models
+                for tl_result in tl_results:
+                    tl_result.update(
+                        {
+                            "mc_iter": mc_iter,
+                            "n_train_pts": n_train_pts,
+                            "fraction_source": fraction_source,
+                            "n_source_pts": len(source_subset),
+                            "n_test_pts": len(target_test),
+                            "source_data_seed": settings.random_seed + mc_iter,
+                        }
+                    )
+                    results.append(tl_result)
+
+                pbar.update(1)
+
+    pbar.close()
+
+    results_df = pd.DataFrame(results)
+
+    return results_df
+
+
+def _sample_source_data(
```

Comment:
any reason why we don't use sklearn's splitting function (which also offers stratification)?